docs/SILProgrammersManual.md

```diff
@@ -159,6 +159,401 @@ rules out PartialApplies is no excuse to conflate applied arguments
+`%load` actually has an empty access path:
+```
+%boxadr = unchecked_take_enum_data_addr %base : $*Enum<T>, #Enum.int!enumelt
+%box = load [take] %boxadr : $*<τ_0_0> { var Int } <T>
+%valadr = project_box %box : $<τ_0_0> { var Int } <T>, 0
+%load = load [trivial] %valadr : $*Int
+```
+
+Storage casts may also occur within an access. This typically results
+from accessors, which perform address-to-pointer
+conversion. Pointer-to-address conversion performs a type cast, and
+could lead to different subobject types corresponding to the same base
+and access path. Access paths still uniquely identify a memory
+location because it is illegal to cast memory to non-layout-compatible
+types on same execution path (without an intervening `bind_memory`).
+
+Address-type phis are prohibited, but because pointer and box types
+may be on the def-use chain, phis may also occur on an access path. A
+phi is only a valid part of an access path if it has no affect on the
+path components. This means that pointer casting and unboxing may
+occur on distinct phi paths, but index offsets and subobject
+projections may not. These rules are currently enforced to a limited
+extent, so it's possible for invalid access path to occur under
+certain conditions.
+
+For example, the following is a valid def-use access chain, with an
+access base defined in `b0`, a memory operation in `b3` and an
+`index_addr` and `struct_element_addr` on the access path:
+
+```
+class A {}
+
+struct S {
+  var field0: Int64
+  var field1: Int64
+}
+
+bb0:
+  %base    = ref_tail_addr %ref : $A, $S
+  %idxproj = index_addr %tail : $*S, %idx : $Builtin.Word
+  %p0 = address_to_pointer %idxproj : $*S to $Builtin.RawPointer
+  cond_br _, bb1, bb2
+
+bb1:
+  %pcopy = copy_value %p0 : $Builtin.RawPointer
+  %adr1  = pointer_to_address [strict] %pcopy : $Builtin.RawPointer to $*S
+  %p1    = address_to_pointer %adr1 : $*S to $Builtin.RawPointer
+  br bb3(%p1 : $Builtin.RawPointer)
+
+bb2:
+  br bb3(%p0 : $Builtin.RawPointer)
+
+bb3(%p3 : $Builtin.RawPointer):
+  %adr3 = pointer_to_address [strict] %p3 : $Builtin.RawPointer to $*S
+  %field = struct_element_addr %adr3 : $*S, $S.field0
+  load %field : $*Int64
+```
+
+### AccessedStorage
+
+`AccessedStorage` identifies an accessed storage location, be it a
+box, stack location, class property, global variable, or argument. It
+is implemented as a value object that requires no compile-time memory
+allocation and can be used as the hash key for that location. Extra
+bits are also available for information specific to a particular
+optimization pass. Its API provides the kind of location being
+accessed and information about the location's uniqueness or whether it
+is distinct from other storage.
+
+Two __uniquely identified__ storage locations may only alias if their
+AccessedStorage objects are identical.
+
+`AccessedStorage` records the "root" SILValue of the access. The root is
+the same as the access base for all storage kinds except global and
```

Comment:
This is actually a very important sentence ("The root is the same as the base, except..."). Maybe it's worth noting that already in the first section.

Reply:
Added earlier
```
#### Access base

The access base is the address that directly identifies the kind of
storage being accessed, such as `alloc_box`, `alloc_stack`,
`global_addr`, `ref_element_addr`, and function arguments (see
`AccessedStorage::Kind`).

The access base is the same as the "root" SILValue for all storage
kinds except global and class storage. Global storage has no root. For
class storage the root is the SILValue that identifies object,
described as the "reference root" above.
```